### PR TITLE
add filetransfer test

### PR DESCRIPTION
**Pull request by fippo**
currently transfers index.html. Should this maybe also transfer some larger files from
https://github.com/webrtc/samples/tree/master/src/content/devices/multi/video ?

### CODE REVIEW

File: src/content/datachannel/filetransfer/js/test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      '/src/content/datachannel/filetransfer/index.html')
+  .then(function() {
+    t.pass('page loaded');
+    // Based on https://saucelabs.com/resources/articles/selenium-file-upload
+    return driver.findElement(webdriver.By.id('fileInput'))
+       .sendKeys(process.cwd() + '/index.html');
+  })
+  .then(function() {
+    // Wait for the received element to be displayed.
+    return driver.wait(function() {
```
Comment (KaptenJansson) [inline src/content/datachannel/filetransfer/js/test.js]: Checkout the [variant](https://github.com/webrtc/samples/commit/3ec2294cadc7e5323f4bb83abc956c3da1ab84e2) I used to fix the flake in the input-output test, yes it does not have a timeout but it does work more cases it seems than using a timeout (seems to trigger on false positives sometimes.).

File: src/content/datachannel/filetransfer/js/test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+test('Filetransfer via Datachannels', function(t) {
+  var driver = seleniumHelpers.buildDriver();
+
+  driver.get('file://' + process.cwd() +
+      '/src/content/datachannel/filetransfer/index.html')
+  .then(function() {
+    t.pass('page loaded');
+    // Based on https://saucelabs.com/resources/articles/selenium-file-upload
+    return driver.findElement(webdriver.By.id('fileInput'))
+       .sendKeys(process.cwd() + '/index.html');
```
Comment (KaptenJansson) [inline src/content/datachannel/filetransfer/js/test.js]: WDYT about using a larger file? Or is that out of the scope as we are just testing that the page can transfer a file?

**Comment by KaptenJansson** [comment]
Note a merge with master is required since I pushed a couple of PR's causing conflict.

File: src/content/datachannel/filetransfer/js/test.js (hunk 2 above)
Comment (fippo) [inline src/content/datachannel/filetransfer/js/test.js]: ah, testing four files of various sizes costs nothing virtually. And this tests that the 16k fragmentation works.

File: src/content/datachannel/filetransfer/js/test.js (hunk 1 above)
Comment (fippo) [inline src/content/datachannel/filetransfer/js/test.js]: I think this needs to check for visibility... let me see...

Comment (KaptenJansson) [inline src/content/datachannel/filetransfer/js/test.js]: You can use the `elementVisible()` method on `webdriver.until` as well, more details [here](http://selenium.googlecode.com/git/docs/api/javascript/namespace_webdriver_until.html).

I just think the combo of `wait()` and `until()` is the way to go in general for elements.

**Comment by KaptenJansson** [comment]
LGTM!

File: src/content/datachannel/filetransfer/js/test.js (hunk 1 above)
Comment (fippo) [inline src/content/datachannel/filetransfer/js/test.js]: elementIsVisible doesn't seem to work for unfortunately. And the current thing doesn't check for visibility, causing things to end early.

**Pull request merged**